be2-scala/src/main/scala/ch/epfl/pop/storage/DbActor.scala

```diff
@@ -83,14 +83,14 @@ final case class DbActor(
   }
 
   @throws [DbActorNAckException]
-  private def writeLaoData(channel: Channel, message: Message): Unit = {
+  private def writeLaoData(channel: Channel, message: Message, address: String): Unit = {
```

Comment:
same comment as above, you could use an Option for address (should be changed in updateWith too then)